python_ta/reporters/json_reporter.py

```diff
@@ -33,8 +33,28 @@ def display_messages(self, layout: BaseLayout) -> None:
             output.append(
                 {
                     "filename": k,
-                    "msgs": [msg.to_dict() for msg in msgs],
+                    "msgs": self._output_messages(msgs),
                 }
             )
 
         self.writeln(json.dumps(output, indent=4))
+
+    def _output_messages(self, msgs: List[NewMessage]) -> List[Dict]:
+        """Returns a list of dictionaries containing formatted error messages."""
+        max_messages = self.linter.config.pyta_number_of_messages
+        unique_msg_names = {msg.message.msg_id for msg in msgs}
```

Comment:
You're on the right track, but can simplify this further by using a dictionary comprehension instead of a set comprehension